library/VTKExtensions/Rendering/vtkF3DRenderer.cxx

```diff
@@ -514,6 +517,28 @@ void vtkF3DRenderer::SetBackground(const double* color)
   this->UpdateTextColor();
 }
 
+//----------------------------------------------------------------------------
+void vtkF3DRenderer::SetLightIntensity(const double intensityFactor)
+{
+  vtkLightCollection* lc = this->GetLights();
+  vtkLight* light;
+  vtkCollectionSimpleIterator it;
+  for (lc->InitTraversal(it); (light = lc->GetNextLight(it));)
+  {
+    double originalIntensity = light->GetIntensity();
+    if (this->originalLightIntensities.count(light))
+    {
+      originalIntensity = this->originalLightIntensities[light];
+    }
+    else
+    {
+      this->originalLightIntensities[light] = originalIntensity;
```

Comment:
I think this map should be cleaned on initialization

Reply:
already done,

Reply:
`.clear()` in `vtkF3DRenderer::Initialize` not good?